File: helparty/src/main/java/com/hamryt/helparty/dto/gym/request/SignUpGymRequest.java
package com.hamryt.helparty.dto.gym.request;

import com.hamryt.helparty.dto.UserType;
import javax.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class SignUpGymRequest {
    
    @NotNull
    private String email;
    
    @NotNull
    private String gymName;
    
    @NotNull
    private String password;
    
    @NotNull
    private String phoneNumber;
    
    @NotNull
    private String addressCode;
    
    @NotNull
    private String addressDetail;
    
    @NotNull
    private UserType userType;
    
    @Builder
    public SignUpGymRequest(
        String email, String gymName, String password, String phoneNumber,
        String addressCode, String addressDetail, UserType userType
    ) {
        this.email = email;
        this.gymName = gymName;
        this.password = password;
        this.phoneNumber = phoneNumber;
        this.addressCode = addressCode;
        this.addressDetail = addressDetail;
        this.userType = userType;
    }
    
}

File: helparty/src/main/java/com/hamryt/helparty/dto/gym/request/SignUpGymRequest.kt
package com.hamryt.helparty.dto.gym.request

import com.hamryt.helparty.dto.UserType
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class SignUpGymRequest(
        var email: @NotEmpty String,
        var gymName: @NotEmpty String,
        var password: @NotEmpty String,
        var phoneNumber: @NotEmpty String,
        var addressCode: @NotEmpty String,
        var addressDetail: @NotEmpty String,
        var userType: @NotNull UserType

) {
    companion object {
        @JvmStatic
        fun create(email: @NotEmpty String, gymName: @NotEmpty String, password: @NotEmpty String,
                   phoneNumber: @NotEmpty String, addressCode: @NotEmpty String, addressDetail: @NotEmpty String,
                   userType: @NotNull UserType) =
                SignUpGymRequest(email, gymName, password, phoneNumber, addressCode, addressDetail, userType)
    }
}